### ISSUE TITLE
SegFault when applying redactions overlapping a transparent image

### ISSUE THREAD

**Issue by JorjMcKie** [pymupdf/PyMuPDF#1824]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
If a redaction rectangle overlaps an image with a mask (= image with transparency) and `page.apply_redactions(images=fitz.PDF_REDACT_IMAGE_PIXELS)` is used (the default), the interpreter crashes with segmentation fault. Verified under Linux and Windows.

## To Reproduce (mandatory)
Execute with the attached reproducer PDF:
```python
import fitz
doc=fitz.open("input.pdf")
page=doc[0]
page.apply_redactions()
```
[input.pdf](https://github.com/pymupdf/PyMuPDF/files/9142817/input.pdf)

## Your configuration (mandatory)
 ```python
In [2]: print(fitz.__doc__)

PyMuPDF 1.20.0: Python bindings for the MuPDF 1.20.1 library.
Version date: 2022-06-27 00:00:01.
Built for Python 3.9 on win32 (64-bit).
```
```python
In [2]: print(fitz.__doc__)

PyMuPDF 1.20.0: Python bindings for the MuPDF 1.20.1 library.
Version date: 2022-06-27 00:00:01.
Built for Python 3.8 on linux (64-bit).
```

## Additional context (optional)
This is also the case for other Python versions - tested up to Python 3.10.
It did work (although with bugs - see comment at the bottom of this) in v 1.19.x.
When recompiling under Windows (Linux yet to be confirmed), things will work again as before.
The error occurs in 

![grafik](https://user-images.githubusercontent.com/8290722/179796170-5ff2d35d-199e-4c73-a990-2759e847c6d4.png)

which invokes this C-code:

![grafik](https://user-images.githubusercontent.com/8290722/179796606-cf3124e2-8dbd-4dd3-b41b-68dcc0fb0d25.png)

-----
Please note that for transparent images, the MuPDF code will produce a PNG image **_without_** transparency.
This is a known problem and does not otherwise interfere with this bug.
Using one of the other options for handling images with redactions continue to work fine.

**Comment by JorjMcKie** [comment]
The error is caused inside MuPDF and will be fixed there.

**Comment by JorjMcKie** [comment]
As a side note:

Once this MuPDF version (and its resp. PyMuPDF version) is out, the older issue of supporting transparent images with redactions will also be solved.

**Comment by julian-smith-artifex-com** [comment]
Fixed in recent release of PyMuPDF-1.20.2.
